### ISSUE TITLE
lmlsb extractmenu failed to find menu

### ISSUE THREAD

**Issue by shy2052** [pmrowla/pylivemaker#76]
* pylivemaker version: 1.0.0
* Python version: 3.7.8
* Operating System: windows 7
* LiveMaker game: Ikisugi M Lesson

### Description

[00000003.zip](https://github.com/pmrowla/pylivemaker/files/5427247/00000003.zip)
The file do have menu text, but `extractmenu` can't find it.
I took a quick look at #60 and #19, it seems menu using `Calc AddArray(_tmp,...)` should be considered as supported by pylm?

### What I Did

part of dumped data from 00000003.lsb:
```
;---------------------------------------
 154: Wait Not(__メッセージ終了) 0 0
 155: Jump 00000003.lsb:157 1 (Label 157: 00000024)
 156: Wait 1 0 0
 157: Label 00000024
 158: Wait ((GetProp("メッセージボックス", 81) | GetProp("メッセージボックス", 138)) & Not(選択実行中)) 0 0
 159: Calc 選択実行中 = 1
 160: VarNew _tmp ParamType.Str  2
 161: VarNew _tmpno ParamType.Int  2
 162: VarNew _tmpid ParamType.Str  2
 163: Calc AddArray(_tmp, "瑞枝さんの所へ行ってみる")
 164: Calc _tmpno = 0
 165: Calc _tmpid = AddDelimiter("\r\n", _tmpid) ++ _tmpno
 166: Calc AddArray(_tmp, "真希に正直に告白する")
 167: Calc _tmpno = 1
 168: Calc _tmpid = AddDelimiter("\r\n", _tmpid) ++ _tmpno
 169: Calc _tmp = Trim(ArrayToString(_tmp))
 170: Calc StringToArray(_tmp, _tmp)
 171: If ArrayToString(_tmp) != ""
 172:     If 1
 173:         Call ノベルシステム\選択メニュー\■選択実行.lsb:0  1 ArrayToString(_tmp) 0 "C" "C" 0 "" "" 1
 174:         Calc 選択実行中 = 0
 175:         Calc StringToArray(_tmpid, _tmpid)
 176:         If 選択番号 >= 0
 177:             Calc 選択番号 = _tmpid[選択番号]
 178:     Else
 179:         Calc 選択値 = _tmp
 180:         Calc 選択番号 = _tmpno
 181:         Calc 選択実行中 = 0
 182:         Calc 右クリックメニュー一時禁止 = 0
 183:         Call ノベルシステム\メッセージボックス\終了.lsb:0  1 
 184:         Wait GetProp("メッセージボックス", 50) != "" 0 0
 185:     ObjDel "VOICE"
 186: VarDel _tmp
 187: VarDel _tmpno
 188: VarDel _tmpid
 189: Jump 00000003.lsb:192 選択値 == "瑞枝さんの所へ行ってみる" (Label 192: 00000025)
 190: Jump 00000003.lsb:195 選択値 == "真希に正直に告白する" (Label 195: 00000026)
 191: Wait 1 0 0
 192: Label 00000025
 193: Call 00000003.lsb:1  1  (Label 1: 変数削除)
 194: Jump 0000000A.lsb:0 1
 195: Label 00000026
 196: Call 00000003.lsb:1  1  (Label 1: 変数削除)
 197: Jump 0000000C.lsb:0 1
 198: Label 00000074
 199: Exit 1
 200: Label 00000075
 201: Jump 00000003.lsb:88 1 (Label 88: 00000019)
 202: Wait 1 0 0
 203: Label 00000076
 204: Call ノベルシステム\シーン回想\■フラグON.lsb:0  1 117
 205: Jump 00000003.lsb:95 1 (Label 95: 0000001A)
 206: Wait 1 0 0
 207: Label 00000077
 208: Call ノベルシステム\シーン回想\■フラグON.lsb:0  1 120
 209: Jump 00000003.lsb:120 1 (Label 120: 0000001E)
 210: Wait 1 0 0
 211: Label 00000078
 212: Jump 00000003.lsb:113 1 (Label 113: 0000001D)
 213: Wait 1 0 0
 214: Label 00000079
 215: Exit 1
 216: Label 0000009C
 217: Jump 00000003.lsb:138 1 (Label 138: 00000021)
 218: Wait 1 0 0
 219: Label 0000009D
 220: Exit 1
 221: Label 0000009E
 222: Call ノベルシステム\シーン回想\■フラグON.lsb:0  1 156
 223: Jump 00000003.lsb:145 1 (Label 145: 00000022)
 224: Wait 1 0 0
```

```
E:\GAME\Ikisugi M Lesson\ikiM\game_files>lmlsb extractmenu --encoding=utf-8-sig 00000003.lsb menu.csv
Extracting 00000003.lsb ...
No menu data found.
```

**Comment by pmrowla** [comment]
Yeah this should be supported, the main issue is just that different LiveMaker releases make small adjustments to how menus were generated, but without example files I couldn't add support for everything at the time menu stuff was implemented in pylm. You'll probably run into the same issue with all the in-game choice menus in your game, I'll try to get a fix done over the weekend.

**Comment by Stefan311** [comment]
seems this older LifeMaker version uses a conditional Jump instead of a unconditional on the last entry of the jumptable.
`190: Jump 00000003.lsb:195 選択値 == "真希に正直に告白する" (Label 195: 00000026)`
instead of
`190: Jump 00000003.lsb:195 1 (Label 195: 00000026)`

**Comment by pmrowla** [comment]
This issue should be resolved in latest (v1.0.1) release
